cmake/modules/platform/toolcfg/gnu.cmake

```diff
@@ -19,10 +19,11 @@
 # SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
 #############################################################################
 
-set(OMR_C_WARNINGS_AS_ERROR_FLAG -Werror)
-set(OMR_CXX_WARNINGS_AS_ERROR_FLAG -Werror)
-set(OMR_NASM_WARNINGS_AS_ERROR_FLAG -Werror)
-
+if(NOT CMAKE_C_COMPILER_IS_OPENXL)
+	set(OMR_C_WARNINGS_AS_ERROR_FLAG -Werror)
+	set(OMR_CXX_WARNINGS_AS_ERROR_FLAG -Werror)
+	set(OMR_NASM_WARNINGS_AS_ERROR_FLAG -Werror)
```

Comment:
Does OpenXL not have an equivalent to `-Werror`?

Reply:
The `-Werror` flag should be compatible with OpenXL. Not sure why it's been excluded for OpenXL here, but I've fixed it

Reply:
At a glance, the errors that occur when using `-Werror` with OpenXL on AIX appear to be pretty simple (e.g.: unused variables, formatting issues), so I think dealing with them and keeping this flag is doable, but will be pretty tedious. @zl-wang is that something we want to save for a future contribution? If so I can open an issue for it and make note of it along with this change as Keith suggested

Reply:
yes, the priorities from higher to lower:
1) allow switch-over to use OpenXL for 0.50 build (?)
2) -Xint performance regression with helps from OpenXL team
3) addressing these warning messages

Reply:
I opened an issue for this here: https://github.com/eclipse-omr/omr/issues/7583 so that it can be addressed later. For now I'll just add a comment referring to it here.